packages/list/src/ListEditor.tsx

```diff
@@ -38,30 +40,45 @@ export function ListEditor(props: ListEditorProps) {
       isEmptyValue={isEmptyListValue}
       field={field}
       isInitiallyDisabled={props.isInitiallyDisabled}>
-      {({ setValue, value, errors, disabled }) => {
-        const valueAsString = (value || []).join(', ');
+      {(childProps) => (
+        <ListEditorInternal {...childProps} direction={direction} isRequired={field.required} />
+      )}
+    </FieldConnector>
+  );
+}
 
-        const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-          const valueAsArray = e.target.value
-            .split(',')
-            .map((item) => item.trim())
-            .filter((item) => item);
-          setValue(valueAsArray);
-        };
+function ListEditorInternal({
+  setValue,
+  value,
+  errors,
+  disabled,
+  direction,
+  isRequired,
+}: FieldConnectorChildProps<ListValue> & { direction: 'rtl' | 'ltr'; isRequired: boolean }) {
+  const [valueState, setValueState] = React.useState(() => (value || []).join(', '));
```

Comment:
minor 💅
```suggestion
  const [valueState, setValueState] = React.useState((value || []).join(', '));
```

Reply:
Theoretically it makes sense to compute the initial state just once, but ofc in this specific case it won't make a big difference in performance

Reply:
I'm not sure if I get your point right.
But just to clarify, there is no difference if you pass initial state as a function or as a value It will be always computed only once on the initial render, isn’t it?

But sure, it's a minor thing and it will not affect performance in any way

Reply:
Not really, if you specify the initial value not wrapped in a function, then it will be computed on every component render. So if we would be doing some expensive calculations to get the initial state, it must be wrapped in the function, then `setState` calls it just once.

Reply:
Ah, I got it now. Thanks @djagya for explaining and sorry for ranting 🙇‍♂️